src/program.ts

```diff
@@ -4569,8 +4585,19 @@ export class Class extends TypedElement {
           implementers.add(extender);
         }
       }
-      nextIface = <Interface | null>nextIface.base;
-    } while (nextIface);
+
+      let nextIfaces = nextIface.interfaces;
+      if (!nextIfaces) continue;
+
+      let stackIndex = ifaceStack.length;
+
+      // Calls the internal ensureCapacity() when run in the bootstrapped compiler:
+      ifaceStack.length = stackIndex + nextIfaces.size;
+  
+      for (let _values = Set_values(nextIfaces), i = 0, k = _values.length; i < k; ++i) {
+        ifaceStack[stackIndex++] = unchecked(_values[i]);
+      }
```

Comment:
It would be easier to understand
```suggestion
      for (let _values = Set_values(nextIfaces), i = 0, k = _values.length; i < k; ++i) {
        ifaceStack.push(unchecked(_values[i]));
      }
```

Reply:
If I remember correctly, I wrote it this way instead of with repeated `.push`es to avoid resizing the array's backing storage multiple times in the bootstrapped version.